test/test_baseparser.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: false
+
+require 'rexml/parsers/baseparser'
+
+module REXMLTests
+  class BaseParserTester < Test::Unit::TestCase
+    include REXML
+
+    N_ELEMENTS = 100
+    N_STRING = 'a' * 50000
+    def build_xml(n_elements)
+      xml = '<?xml version="1.0"?><root>'
+
+      n_elements.times do |i|
+        xml << '<child >'
+        xml <<  N_STRING
+        xml <<  '</child>'
+      end
+      xml << '</root>'
+    end
+
+    def test_parse_large_xml
+      xml = build_xml(N_ELEMENTS)
+      parser = REXML::Parsers::BaseParser.new(xml)
+      while parser.has_next?
+        parser.pull
+        assert_compare REXML::Source::SCANNER_RESET_SIZE + N_STRING.size, ">", parser.position
+      end
+    end
```

Comment:
How about doing them all in the test because `build_xml` will not be reused? Something like:

```ruby
def test_large_xml
  large_text = "a" * 100_000
  xml = <<-XML
    <?xml version="1.0"?>
    <root>
      <child>#{large_text}</child>
      <child>#{large_text}</child>
    </root>
  XML

  parser = REXML::Parsers::BaseParser.new(xml)
  while parser.has_next?
    parser.pull
  end
  assert do
    source.position < xml.bytesize
  end
end
```

Reply:
OK.
I See.